libs/api/src/main/java/de/cyface/api/model/ClassifiedSegment.java

```diff
@@ -0,0 +1,434 @@
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Cyface Data Collector is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the Cyface Data Collector. If not, see <http://www.gnu.org/licenses/>.
+ */
+package de.cyface.api.model;
+
+import de.cyface.model.Modality;
+import org.apache.commons.lang3.Validate;
+
+import java.time.OffsetDateTime;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.Consumer;
+
+import static de.cyface.api.model.Json.jsonArray;
+import static de.cyface.api.model.Json.jsonKeyValue;
+import static de.cyface.api.model.Json.jsonObject;
+
+/**
+ * Class which represents result elements from the surface pipeline: classified road segments.
+ * <p>
+ * The data in this class is generated from {@code LocationInSegment} data.
+ *
+ * @author Armin Schnabel
+ */
+@SuppressWarnings("unused") // Part of the API
+public class ClassifiedSegment {
+
+    /**
+     * The current version of the format of this segment class.
+     */
+    public final String CURRENT_VERSION = "1.0.0";
+    /**
+     * The database identifier of the segment.
+     */
+    final private String oid;
+    /**
+     * {@code True} of this segment is orientated in the same direction as the formal direction of the underlying OSM
+     * way id or {@code false} if it's orientated in the opposite direction.
+     */
+    final private boolean forward;
+    /**
+     * The geometry of this segment in the GeoJSON format, i.e. containing a `type` attribute with `LineString` as value
+     * and the `coordinates` attribute with an array containing the nodes of this segment, as loaded from the database.
+     */
+    final private Geometry geometry;
+    /**
+     * The length of this segment in meters.
+     */
+    final private double length;
+    /**
+     * The {@link Modality} this segment was recorded with.
+     */
+    final private Modality modality;
+    /**
+     * The OSM id of the node where the underlying OSM way of this segment starts.
+     */
+    final private long vnk;
+    /**
+     * The OSM id of the node where the underlying OSM way of this segment ends.
+     */
+    final private long nnk;
+    /**
+     * The meter this segment start at compared the full way from `vnk` to `nnk`. I.e. the travel-distance to `vnk`.
+     * <p>
+     * FIXME: Should we not also store the max_length (100) or segmentation (100)?
+     * This helps the client to show "meter 100-200" and to store segments for 25, 100 segmentation at the same time.
```

Comment:
I meant: we should also store the `max_length` of a segment to differentiate between e.g. `50` and `100` meter segments. Did we discuss this before? If not, do I just create a task for this and link the task in this comment? Or do we just not support this feature yet (i.e. only one segment length per user? Then I'll drop this comment)